pybamm/expression_tree/concatenations.py

```diff
@@ -30,6 +30,7 @@ def __init__(self, *children, name=None, check_domain=True):
         else:
             domain = []
         super().__init__(name, children, domain=domain)
+        self.cached_children = self.children
```

Comment:
is there a reason to only do this for binary operators and concatenations? Could simply do this for the base class `Symbol`, and then overload the `children` attribute to return `cached_children`?

Reply:
Ok, no reason, just hadn't thought of doing it that way. So if I understand correctly, we could just add a line in `Symbol.__init__` that does something like the following?
```
self.children = anytree.NodeMixin(self, children)
```